app/services/event.rb

```diff
@@ -34,15 +34,13 @@ def base_data
   end
 
   ##
-  # For json objects or hashes passed to Event#trigger as values in the `data` param, such as
-  # Subscription#search_criteria or Feedback#search_criteria, we should format these as json for BigQuery,
-  # rather than strings, for easier data manipulation by Performance Analysis.
-  # Floats, Integers and Arrays, such as job alert Feedbacks with a job_alert_vacancy_ids attribute,
-  # should remain as they are.
+  # For json objects or hashes passed to Event#trigger as values in the `data` param, such as Feedback#search_criteria,
+  # we should format these as json for BigQuery, rather than strings, for easier manipulation by Performance Analysis.
+  # Floats and Integers should remain as they are. Do not pass Arrays to BigQuery: otherwise, it will give the
+  # error "Array specified for non repeated field" when the array has length of 1.
   # @param [Object] value Any value in the data passed to the event.
   def formatted_value(value)
     return value if value.is_a?(Float) || value.is_a?(Integer)
```

Comment:
We can still pass arrays, but if the array is length 1 we should just pass the value. This stops us sending arrays altogether though doesn't it?

Reply:
Yes, but I reckon it would be confusing to be storing an attribute in multiple formats: both single values and stringified arrays.

Reply:
Compare: https://github.com/DFE-Digital/teaching-vacancies/blob/338d3b0a9553bbe5dec4dc0dd78bb1e6efac6f75/lib/export_tables_to_big_query.rb#L109

Reply:
Ok. Can you update the comment to say that Arrays are converted to strings rather than they're not sent?

Reply:
Yeah